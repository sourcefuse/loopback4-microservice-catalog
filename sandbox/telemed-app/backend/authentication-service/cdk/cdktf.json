{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "b60ac122-4548-411b-a6a9-948a785adbfc",
  "sendCrashReports": "false",
  "terraformProviders": [],
  "terraformModules": [
    {
      "name": "redis",
      "source": "cloudposse/elasticache-redis/aws"
    }
  ],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
