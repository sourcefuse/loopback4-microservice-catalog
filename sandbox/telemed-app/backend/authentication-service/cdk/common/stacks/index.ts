export * from './lambda.stack';
export * from './migration.stack';
export * from './redis.stack';
