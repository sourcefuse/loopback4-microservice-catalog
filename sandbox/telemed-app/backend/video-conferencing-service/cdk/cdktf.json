{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "7da9330e-d08e-43cb-967a-62f41d6fac01",
  "sendCrashReports": "false",
  "terraformProviders": [],
  "terraformModules": [
    {
      "name": "redis",
      "source": "cloudposse/elasticache-redis/aws"
    }
  ],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
