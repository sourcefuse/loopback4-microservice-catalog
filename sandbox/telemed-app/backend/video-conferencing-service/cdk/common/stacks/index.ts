export * from "./lambda.stack";
export * from "./migration.stack";
