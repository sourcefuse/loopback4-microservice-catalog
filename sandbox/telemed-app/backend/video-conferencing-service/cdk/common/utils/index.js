'use strict';
// sonarignore:start
let __createBinding =
  (this && this.__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        // sonarignore:end
        if (k2 === undefined) k2 = k;
        let desc = Object.getOwnPropertyDescriptor(m, k);
        if (
          !desc ||
          ('get' in desc ? !m.__esModule : desc.writable || desc.configurable)
        ) {
          desc = {
            enumerable: true,
            get: function () {
              return m[k];
            },
          };
        }
        Object.defineProperty(o, k2, desc);
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });
// sonarignore:start
let __exportStar =
  (this && this.__exportStar) ||
  function (m, exports) {
    // sonarignore:end
    for (let p in m)
      if (p !== 'default' && !Object.hasOwn.call(exports, p))
        __createBinding(exports, m, p);
  };
Object.defineProperty(exports, '__esModule', {value: true});
__exportStar(require('./helper'), exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJpbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7Ozs7O0FBQUEsMkNBQXlCIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSAnLi9oZWxwZXInO1xuIl19
