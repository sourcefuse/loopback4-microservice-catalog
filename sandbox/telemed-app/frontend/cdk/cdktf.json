{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "116578e3-9930-4a16-8805-b23da1f7d156",
  "sendCrashReports": "false",
  "terraformProviders": [],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
