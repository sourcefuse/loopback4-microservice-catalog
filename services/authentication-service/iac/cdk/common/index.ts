export * from './stacks';
export * from './utils';
