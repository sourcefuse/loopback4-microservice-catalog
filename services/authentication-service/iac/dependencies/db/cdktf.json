{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "ba437304-bcc3-4140-9a42-e0d3d2cb5725",
  "sendCrashReports": "false",
  "terraformProviders": [],
  "terraformModules": [
    {
      "name": "dbModule",
      "source": "git::https://github.com/sourcefuse/terraform-aws-ref-arch-db//"
    }
  ],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
