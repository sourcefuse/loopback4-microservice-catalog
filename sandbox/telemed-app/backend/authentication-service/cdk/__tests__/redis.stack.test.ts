// https://cdk.tf/testing
import { Testing } from 'cdktf';
import 'cdktf/lib/testing/adapters/jest'; // Load types for expect matchers
import { RedisStack } from '../common';

expect.addSnapshotSerializer({
  test: val => typeof val === 'string',
  print: val => {
    if (typeof val === 'string') {
      const newVal = val.replace(/[A-Z0-9]{32}/g, 'ABC123');
      return `${newVal}`;
    }
    return '';
  },
});



describe('My CDKTF Application', () => {
  let redisStack: RedisStack;
  let stack: string;

  beforeEach(() => {
    const app = Testing.app();
    redisStack = new RedisStack(app, 'test',{
        environment:'test',
        namespace:'test'
    });
    stack = Testing.synth(redisStack);
  });

  it('should match snapshot test', () => {
    expect(stack).toMatchSnapshot();
  });

  it('check if the produced terraform configuration is valid', () => {
    expect(Testing.fullSynth(redisStack)).toBeValidTerraform();
  });

});


