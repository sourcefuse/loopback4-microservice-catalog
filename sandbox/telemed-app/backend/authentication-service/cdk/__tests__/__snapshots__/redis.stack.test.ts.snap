// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My CDKTF Application should match snapshot test 1`] = `
{
  "data": {
    "aws_security_groups": {
      "redis_user_sg": {
        "filter": [
          {
            "name": "tag:redis-user",
            "values": [
              "yes"
            ]
          },
          {
            "name": "vpc-id",
            "values": [
              "\${data.aws_vpc.vpc.id}"
            ]
          }
        ]
      }
    },
    "aws_subnet": {
      "subnet": {
        "for_each": "\${toset(toset(data.aws_subnets.private.ids))}",
        "id": "\${each.value}"
      }
    },
    "aws_subnets": {
      "private": {
        "filter": [
          {
            "name": "tag:Name",
            "values": [
              "test-test-privatesubnet-private-undefineda",
              "test-test-privatesubnet-private-undefinedb"
            ]
          },
          {
            "name": "vpc-id",
            "values": [
              "\${data.aws_vpc.vpc.id}"
            ]
          }
        ]
      }
    },
    "aws_vpc": {
      "vpc": {
        "filter": [
          {
            "name": "tag:Name",
            "values": [
              "test-test-vpc"
            ]
          }
        ]
      }
    }
  },
  "module": {
    "redis": {
      "allowed_security_group_ids": "\${data.aws_security_groups.redis_user_sg.ids}",
      "apply_immediately": true,
      "at_rest_encryption_enabled": true,
      "description": "Elasticache redis instance for ref arch devops infra",
      "environment": "test",
      "namespace": "test",
      "source": "cloudposse/elasticache-redis/aws",
      "subnets": "\${data.aws_subnets.private.ids}",
      "transit_encryption_enabled": true,
      "vpc_id": "\${data.aws_vpc.vpc.id}"
    }
  },
  "provider": {
    "aws": [
      {
        "assume_role": [
          {
          }
        ]
      }
    ]
  },
  "resource": {
    "aws_security_group": {
      "ec_security_group": {
        "description": "Security Group for ElastiCache redis users",
        "egress": [
          {
            "cidr_blocks": "\${[for s in data.aws_subnet.subnet : s.cidr_block]}",
            "description": "Rule to allow Redis users to access the Redis cluster",
            "from_port": 0,
            "ipv6_cidr_blocks": null,
            "prefix_list_ids": null,
            "protocol": "TCP",
            "security_groups": null,
            "self": null,
            "to_port": 6379
          }
        ],
        "name": "test-test-Redis-user",
        "tags": {
          "Name": "test-test-Redis-user",
          "redis-user": "yes"
        },
        "vpc_id": "\${data.aws_vpc.vpc.id}"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "4.51.0"
      }
    }
  }
}
`;
